### DISCUSSION TITLE
Helix to open multiple buffers for the same file if it's passed in multiple times

### DISCUSSION THREAD

**Discussion by nik-rev** [helix-editor/helix#12184] [Ideas]
Currently, you can add `:line_number` at the end of a file name to open it at a specific line

```sh
hx LICENSE:4
```

The above opens `LICENSE` file at line 4. `LICENSE:6:3` would open it at line 6, column 3.

What if I want to open LICENSE file at line 4 and line 6 column 3, two buffers for the same file?

```sh
hx LICENSE:4 LICENSE:6:3
```

It won't work, Helix will just open one buffer for one of the inputs. What would be nice, if it instead opened a separate buffer for each. So the above command will open two buffers, one with the cursor at line 4 of LICENSE and the other with the cursor at line 6 of LICENSE.

This is going to be **very useful** in a lot of situations. For example, consider a tool like `cspell` which outputs in the following format:

```sh
README.md:170:229 - Unknown word (turles)
README.md:172:50 - Unknown word (inacurate)
README.md:173:25 - Unknown word (advsed)
README.md:182:51 - Unknown word (patricular)
README.md:190:81 - Unknown word (sofwtare)
README.md:193:291 - Unknown word (neihter)
README.md:201:289 - Unknown word (auhtor)
```

If only I could run a command like this, which I could compute from the above output:

```sh
hx README.md:170:229 README.md:172:50 README.md:173:25 README.md:182:51 README.md:190:81 README.md:193:291 README.md:201:289
```

Which would open 7 buffers, with cursor exactly at each line and column. Then I could do `gn` and `gp` to cycle between the spelling errors and fix them. Thoughts?

**Comment by oxcrow** [comment]
I want a similar feature but for a different reason.

Files can become large. I want to open the same file in multiple tabs (but in same buffer), so I can work on different sections of the code in different tabs.

For example, In one tab I can write unit tests, in another I can edit the function I am testing.

**Reply by nik-rev** [reply]
> I want a similar feature but for a different reason.
> 
> Files can become large. I want to open the same file in multiple tabs (but in same buffer), so I can work on different sections of the code in different tabs.
> 
> For example, In one tab I can write unit tests, in another I can edit the function I am testing.

Feel free to create a 2nd discussion. This one can be closed because  it was implemented in https://github.com/helix-editor/helix/pull/12192
